### ISSUE TITLE
false positive on while_immutable_condition

### ISSUE THREAD

**Issue by dimbleby** [rust-lang-nursery/rust-clippy#2540]
Run clippy on [dimbleby/mio-extras](https://github.com/dimbleby/mio-extras), get this warning:

```
$ cargo +nightly clippy
lib: mio-extras
   Compiling cfg-if v0.1.2
   Compiling libc v0.2.39
   Compiling lazycell v0.6.0
   Compiling slab v0.4.0
   Compiling log v0.4.1
   Compiling net2 v0.2.32
   Compiling iovec v0.1.2
   Compiling mio v0.6.14
   Compiling mio-extras v2.0.3 (file:///data/dch/mio-extras)
warning: Variable in the condition are not mutated in the loop body. This either leads to an infinite or to a never running loop.
   --> src/timer.rs:266:9
    |
266 | /         while self.tick <= target_tick {
267 | |             let curr = self.next;
268 | |
269 | |             trace!("ticking; curr={:?}", curr);
...   |
307 | |             }
308 | |         }
    | |_________^
    |
    = note: #[warn(while_immutable_condition)] on by default
    = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.188/index.html#while_immutable_condition

    Finished dev [unoptimized + debuginfo] target(s) in 4.28 secs
```

This is incorrect - the loop progresses [here](https://github.com/dimbleby/mio-extras/blob/3341fa96741c214e8b79500e96606a6b83192180/src/timer.rs#L272).

```
$ cargo +nightly clippy --version
0.0.188
```
